### PR TITLE
Added fullscreen support on all OS

### PR DESCRIPTION
**Pull request by Rsedaikin**
Added native fullscreen support for macos and exclusive fullscreen for windows and linux. Also did some refactoring.

### CODE REVIEW

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/HardwareLayer.kt [hunk 1]
```diff
@@ -15,20 +16,35 @@ open class HardwareLayer : Canvas(), Drawable {
     override fun paint(g: Graphics) {
         display()
     }
+
+    private val platformOperations by lazy { getPlatformOperations() }
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/HardwareLayer.kt:20]: Make that a top-level property, and get rid of property here. It is always the same

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/OpenGLApi.kt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package org.jetbrains.skiko
 
+import org.jetbrains.skiko.Library
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/OpenGLApi.kt:3]: Remove it.

File: skiko/src/jvmMain/objectiveC/macos/drawlayer.m [hunk 3]
```diff
@@ -182,10 +184,33 @@ - (void) disposeLayer:(JNIEnv *) env
     self.glLayer.canvasGlobalRef = NULL;
     self.glLayer = NULL;
     self.container = NULL;
+    self.window = NULL;
+}
+
+- (BOOL) isFullScreen {
+    NSUInteger masks = [self.window styleMask];
+    if ((masks & NSWindowStyleMaskFullScreen) != 0) {
```
Comment (olonho) [inline skiko/src/jvmMain/objectiveC/macos/drawlayer.m:192]: `return (masks & NSWindowStyleMaskFullScreen) != 0`

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/HardwareLayer.kt (hunk 1 above)
Comment (Rsedaikin) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skiko/HardwareLayer.kt]: Done

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/OpenGLApi.kt (hunk 2 above)
Comment (Rsedaikin) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skiko/OpenGLApi.kt]: Done

File: skiko/src/jvmMain/objectiveC/macos/drawlayer.m (hunk 3 above)
Comment (Rsedaikin) [reply skiko/src/jvmMain/objectiveC/macos/drawlayer.m]: Done

**Pull request closed without merging**